## Task 17 — Revenue per Customer by Country

### Business Question

Which countries generate the highest revenue per customer?

### Analysis

Group customers by country, calculate the number of customers and total monthly revenue for each country, then calculate revenue per customer and sort the results in descending order by this metric.

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
total_revenue_by_country = starlink.groupby("country").agg(
    users_count = ("unique_id","count"),
    total_revenue = ("monthly_fee","sum")
)
total_revenue_by_country["revenue_per_customer"] = total_revenue_by_country["total_revenue"] / total_revenue_by_country["users_count"]
total_revenue_by_country_sorted = total_revenue_by_country.sort_values(
    "revenue_per_customer",
    ascending = False
)
print(total_revenue_by_country_sorted)

                users_count  total_revenue  revenue_per_customer
country                                                         
Japan                   973       117710.0            120.976362
Canada                  997       119630.0            119.989970
Spain                  1019       122100.0            119.823356
United States           954       114250.0            119.758910
Australia              1030       122360.0            118.796117
France                  996       117780.0            118.253012
Ukraine                1032       121760.0            117.984496
Brazil                  996       115680.0            116.144578
United Kingdom          977       113450.0            116.120778
Germany                1027       118550.0            115.433301


### Result

Japan has the highest revenue per customer at 120.98, followed by Canada at 119.99 and Spain at 119.82. Germany has the lowest revenue per customer at 115.43.

### Business Insight

Revenue per customer is relatively consistent across countries, with only a moderate difference between the highest and lowest values. Japan generates the highest revenue per customer, while Germany generates the lowest.